compiler/noirc_evaluator/src/ssa/function_builder/mod.rs

```diff
@@ -442,37 +442,44 @@ impl FunctionBuilder {
     /// within the given value. If the given value is not an array and does not contain
     /// any arrays, this does nothing.
     pub(crate) fn increment_array_reference_count(&mut self, value: ValueId) {
-        self.update_array_reference_count(value, true);
+        self.update_array_reference_count(value, true, true);
```

Comment:
1. Is this always called when `found_ref == true` or does `found_ref` mean something different than "a reference has been found"?
2. I'd prefer these `true`'s to be labeled, at least when used outside of the helper functions

Reply:
I struggled naming this parameter. For parameters we only want to issue rc instructions after we find a reference type and recur, hence `found_ref`, but for other cases like assigning to a `let` binding we always want to inc_rc so `found_ref` is just overridden to true. I'm going to change this to remove the need for the parameter.

Reply:
Removed